lms/djangoapps/certificates/badge_handler.py

```diff
@@ -0,0 +1,134 @@
+"""
+BadgeHandler object-- used to award Badges to users who have completed courses.
+"""
+import hashlib
+import logging
+import requests
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.core.urlresolvers import reverse
+from lazy.lazy import lazy
+from requests.packages.urllib3.exceptions import HTTPError
+from certificates.models import BadgeTracker
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BadgeHandler(object):
+    """
+    The only properly public method of this class is 'award'. If an alternative object is created for a different
+    badging service, the other methods don't need to be reproduced.
+    """
```

Comment:
We don't have a particular style guide for docstrings. We require you to have them and that pylint doesn't complain about the line length. I've seen quite a bit of variance on this but it hasn't seemed to be something that upstream has wanted to make an issue of, in the scheme of everything else that needs to be focused on.

The most important thing is that is has docstrings to begin with, after all. :)